src/printer_yin.c

```diff
@@ -0,0 +1,1557 @@
+                    ypr_unsigned(ctx, LYEXT_SUBSTMT_MAX, 0, rpl->exts, rpl->max);
+                } else {
+                    ypr_substmt(ctx, LYEXT_SUBSTMT_MAX, 0, "unbounded", rpl->exts);
+                }
+            }
+        } else if (elem->mod == LYS_DEV_DELETE) {
+            del = (struct lysp_deviate_del*)elem;
+            ly_print(ctx->out, "delete\">\n");
+            LEVEL++;
+
+            yprp_extension_instances(ctx, LYEXT_SUBSTMT_SELF, 0, del->exts, NULL, 0);
+            ypr_substmt(ctx, LYEXT_SUBSTMT_UNITS, 0, del->units, del->exts);
+            LY_ARRAY_FOR(del->musts, v) {
+                yprp_restr(ctx, &del->musts[v], "must", "condition", NULL);
+            }
+            LY_ARRAY_FOR(del->uniques, v) {
+                ypr_substmt(ctx, LYEXT_SUBSTMT_UNIQUE, v, del->uniques[v], del->exts);
+            }
+            LY_ARRAY_FOR(del->dflts, v) {
+                ypr_substmt(ctx, LYEXT_SUBSTMT_DEFAULT, v, del->dflts[v], del->exts);
+            }
+        }
+
+        LEVEL--;
+        ypr_close(ctx, "deviate", 1);
+    }
+
+    LEVEL--;
+    ypr_close(ctx, "deviation", 1);
+}
+
+/**
+ * @brief Minimal print of a schema.
+ *
+ * To print
+ * a) compiled schema when it is not compiled or
+ * b) parsed when the parsed form was already removed
+ */
+static LY_ERR
+ypr_missing_format(struct ypr_ctx *ctx, const struct lys_module *module)
+{
+    /* module-header-stmts */
+    if (module->version) {
+        if (module->version) {
+            ypr_substmt(ctx, LYEXT_SUBSTMT_VERSION, 0, module->version == LYS_VERSION_1_1 ? "1.1" : "1", NULL);
+        }
+    }
+    ypr_substmt(ctx, LYEXT_SUBSTMT_NAMESPACE, 0, module->ns, NULL);
+    ypr_substmt(ctx, LYEXT_SUBSTMT_PREFIX, 0, module->prefix, NULL);
+
+    /* meta-stmts */
+    if (module->org || module->contact || module->dsc || module->ref) {
+        ly_print(ctx->out, "\n");
+    }
+    ypr_substmt(ctx, LYEXT_SUBSTMT_ORGANIZATION, 0, module->org, NULL);
+    ypr_substmt(ctx, LYEXT_SUBSTMT_CONTACT, 0, module->contact, NULL);
+    ypr_substmt(ctx, LYEXT_SUBSTMT_DESCRIPTION, 0, module->dsc, NULL);
+    ypr_substmt(ctx, LYEXT_SUBSTMT_REFERENCE, 0, module->ref, NULL);
+
+    /* revision-stmts */
+    if (module->revision) {
+        ypr_open(ctx, "revision", "date", module->revision, -1);
+    }
+
+    LEVEL--;
+    ypr_close(ctx, "module", 1);
+    ly_print_flush(ctx->out);
+
+    return LY_SUCCESS;
+}
+
+static void
+ypr_xmlns(struct ypr_ctx *ctx, const struct lys_module *module)
+{
+    unsigned int u;
+    char *space = "        ";
+
+    ly_print(ctx->out, "%s%*sxmlns=\"%s\"", space, INDENT, YIN_NS_URI);
+    ly_print(ctx->out, "\n%s%*sxmlns:%s=\"%s\"", space, INDENT, module->prefix, module->ns);
+
+    struct lysp_module *modp = module->parsed;
+
+    LY_ARRAY_FOR(modp->imports, u){
+        ly_print(ctx->out, "\n%s%*sxmlns:%s=\"%s\"", space, INDENT, modp->imports[u].prefix, modp->imports[u].module->ns);
+    }
+}
+
+
+struct ext_substmt_info_s stmt_attr_info[] = {
```

Comment:
Not sure, @Dajvid isn't this information already somewhere in the YIN parser? Or is it just hard-coded in the YIN parser's functions?

Reply:
In YIN parser, this information is hardcoded in functions that are responsible for parsing of particular elements. At that time I didn't find it useful to have it separated, but now, when it's also needed in yin printer, it would be better to have it all at one place in format that is suitable for both printer and parser. Later I definitely could make some changes in YIN parser to have this information shared, but for that purpose, it would be much better to use enum yin_argument (currently defined in parser_yin.h) for identification of argument instead of string (for example YIN_ARG_NAME instead of "name") otherwise, I would either have to again, slightly modify this table and thus duplicate this information in YIN parser or unnecessarily compare strings when processing each argument multiple times.

So I would suggest to slightly change this table to use mentioned enumeration (there is also macro yin_attr2str that maps enum to string (yin_attr2str(YIN_ARG_NAME) == "name") and when this pull request will be merged I will change YIN parser to also use information from this table.

Reply:
Yes, I see some in parser. But they are only half of them. I don't want to modify them in this PR in case of making too many changes in yin parser. Should I modify them here?

Reply:
I would actually let it as it is for now. Looking more into it (checking where and for what the structure is used) make me feel that the change will be more complex and it would be better to make it separated. It seems that actually the name of the structure itself (`ext_substmt_info_s`) should be changed to reflect its changed purpose (because its original purpose was slightly different).

So I agree with the proposed changes, I would just postopne the changes to a separated task.